src/12-fine-tuning/lib/data.ts

```diff
@@ -0,0 +1,87 @@
+import fs from "fs"
+import path from "path"
+import dedent from "dedent"
+import { SELECTED_INPUT } from "12-fine-tuning/01-medium"
+
+const SYSTEM_PROMPT =
+  "Your task is to identify the medium (aka 'medium type') of an artwork based on a description of the materials within it. If you cannot infer the medium type then answer with the string <unknown>."
+
+/**
+ * Takes raw data from Gravity of the form…
+ *
+ * ```json
+ * { "category": "Painting", "medium": "Oil on Wood" }
+ * { "category": "Sculpture", "medium": "Stoneware" }
+ * ```
+ * …and turns it into properly formatted training and validation data for the model.
+ *
+ * The raw data files are expected in the `data/raw` directory, in JSONL format.
+ */
+export async function prepareData() {
+  const dataPath = path.resolve(
+    __dirname,
+    "..",
+    "data",
+    "raw",
+    `${SELECTED_INPUT}.jsonl`
+  )
+  const data = fs.readFileSync(dataPath, "utf-8")
+
+  const trainingSetPath = path.resolve(
+    __dirname,
+    "..",
+    "data",
+    "training",
+    "medium-training-data.jsonl"
+  )
+  const trainingSet = fs.createWriteStream(trainingSetPath)
+
+  const validationSetPath = path.resolve(
+    __dirname,
+    "..",
+    "data",
+    "training",
+    "medium-validation-data.jsonl"
+  )
+  const validationSet = fs.createWriteStream(validationSetPath)
+
+  const documents = data.split("\n")
+  const trainingSetSize = Math.round(0.8 * documents.length)
+  const validationSetSize = documents.length - trainingSetSize
```

Comment:
Keeping 80% of the data for training, and the remaining 20% for validation. The validation data is used by OpenAI during the training process, though documentation for this is scarce.